staticaddr/loopin/actions.go

```diff
@@ -804,152 +805,65 @@ func (f *FSM) FetchSignPushSweeplessSweepTxAction(ctx context.Context,
-	if err != nil {
-		err = fmt.Errorf("unable to convert high fee server "+
-			"nonces: %w", err)
-
-		return f.HandleError(err)
-	}
-
-	// Extremely high fee.
-	extremeFeeRate := chainfee.SatPerKWeight(
-		fetchResp.ExtremeFeeInfo.FeeRate,
-	)
-	serverExtremeNonces, err := toNonces(
-		fetchResp.ExtremeFeeInfo.Nonces,
-	)
-	if err != nil {
-		err = fmt.Errorf("unable to convert extremely high fee "+
-			"server nonces: %w", err)
-
-		return f.HandleError(err)
-	}
-
-	// Standard sessions.
-	sessions, nonces, err := f.loopIn.createMusig2Sessions(
-		ctx, f.cfg.Signer,
-	)
-	if err != nil {
-		return f.HandleError(err)
-	}
-	clientNonces, err := toNonces(nonces)
-	if err != nil {
-		return f.HandleError(err)
-	}
-
-	// High fee sessions.
-	highFeeSessions, highFeeClientNonces, err :=
-		f.loopIn.createMusig2Sessions(ctx, f.cfg.Signer)
-
-	if err != nil {
-		return f.HandleError(err)
-	}
-	highClientNonces, err := toNonces(highFeeClientNonces)
-	if err != nil {
-		return f.HandleError(err)
-	}
-
-	// Extremely high sessions.
-	extremeSessions, extremeClientNonces, err :=
-		f.loopIn.createMusig2Sessions(ctx, f.cfg.Signer)
-
-	if err != nil {
-		return f.HandleError(err)
-	}
-	extremelyHighClientNonces, err := toNonces(extremeClientNonces)
-	if err != nil {
-		return f.HandleError(err)
-	}
+	clientResponse := make(map[uint64]*looprpc.ClientSweeplessSigningInfo)
 
-	// Create standard fee.
-	sweepTx, err := f.loopIn.createSweeplessSweepTx(address, feeRate)
-	if err != nil {
-		err = fmt.Errorf("unable to create sweepless sweep tx: %w", err)
-		return f.HandleError(err)
-	}
-
-	// Create high fee.
-	highFeeSweepTx, err := f.loopIn.createSweeplessSweepTx(
-		address, highFeeRate,
+	var (
+		nonces              [][]byte
+		clientNonces        [][musig2.PubNonceSize]byte
+		serverNonces        [][musig2.PubNonceSize]byte
+		sessions            []*input.MuSig2SessionInfo
+		sweepTx             *wire.MsgTx
+		sweeplessClientSigs [][]byte
 	)
-	if err != nil {
-		err = fmt.Errorf("unable to create high fee sweepless sweep "+
-			"tx: %w", err)
-
-		return f.HandleError(err)
-	}
-
-	// Create extremely high fee.
-	extremelyHighFeeSweepTx, err := f.loopIn.createSweeplessSweepTx(
-		address, extremeFeeRate,
-	)
-	if err != nil {
-		err = fmt.Errorf("unable to create extremely high fee "+
-			"sweepless sweep tx: %w", err)
-
-		return f.HandleError(err)
-	}
+	for feeRate, serverInfo := range fetchResp.FeeRateToNonces {
+		satPerKW := chainfee.SatPerKWeight(feeRate)
+		serverNonces, err = toNonces(serverInfo.Nonces)
+		if err != nil {
+			err = fmt.Errorf("unable to convert server nonces: %w",
+				err)
+			return f.HandleError(err)
+		}
 
-	// Sign standard.
-	sweeplessClientSigs, err := f.loopIn.signMusig2Tx(
-		ctx, sweepTx, f.cfg.Signer, sessions, serverNonces,
-	)
-	if err != nil {
-		err = fmt.Errorf("unable to sign sweepless sweep tx: %w", err)
-		return f.HandleError(err)
-	}
+		sessions, nonces, err = f.loopIn.createMusig2Sessions(
```

Comment:
I think, resources represented by `sessions` leak in the signer. We should close them. It is not related to this PR, therefore let's add TODO here to clean sessions in the future.